### PR TITLE
add aspnetcore adapter and samples

### PR DESCRIPTION
**Pull request by lust4life**
migrate repo from https://github.com/logary/AspNetCore into logary

### CODE REVIEW

**Comment by lust4life** [comment]
and the failed travis ci caused by `The job exceeded the maximum log length, and has been terminated.` , this may be a mis-compare changes by `paket`, see https://github.com/fsprojects/Paket/pull/3464 

Temporarily ignore checks for ci passed, after pr above have been merged and update paket, can revert this ignore check changes

**Comment by lust4life** [comment]
@haf hi,
can you help merge and make a release, or are there any concerns from your point of view ?

**Comment by haf** [comment]
@lust4life It's just been a lot of xmas here, away from the laptop. Let me read through it.

File: src/adapters/Logary.Adapters.AspNetCore/Extensions.fs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+open Logary.Configuration
+open Microsoft.Extensions.Logging
+open System.Runtime.CompilerServices
+
+
+[<AutoOpen;Extension>]
+module ILoggingBuilderExtensions =
+
+    [<Extension>]
```
Comment (haf) [inline src/adapters/Logary.Adapters.AspNetCore/Extensions.fs]: Better to use `CompiledName` here, than to use a non-standard method signature for this project; that way, if you use this code from C#, you'll have the NameCasedLikeThis.

File: examples/aspnetcore/AspNetCore.CSharp/appsettings.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿{
+  "Logging": {
+    "LogLevel": {
+      "Default": "Warning"
```
Comment (haf) [inline examples/aspnetcore/AspNetCore.CSharp/appsettings.json]: Shouldn't this be lower? To let Logary handle the levels?

File: examples/aspnetcore/AspNetCore.CSharp/Properties/launchSettings.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
+﻿{
```
Comment (haf) [inline examples/aspnetcore/AspNetCore.CSharp/Properties/launchSettings.json]: Can this be launched on macOS? (just curious)

Comment (lust4life) [reply examples/aspnetcore/AspNetCore.CSharp/Properties/launchSettings.json]: yes, `dotnet run` this proj, then browser `http://localhost:5000/api/TestScope`

File: examples/aspnetcore/AspNetCore.CSharp/appsettings.json (hunk 2 above)
Comment (lust4life) [reply examples/aspnetcore/AspNetCore.CSharp/appsettings.json]: there are two ways, let aspnet core pass all message to logary or  vice versa, here is aspnet core logging way to handler rules and levels.

File: src/adapters/Logary.Adapters.AspNetCore/Extensions.fs (hunk 1 above)
Comment (lust4life) [reply src/adapters/Logary.Adapters.AspNetCore/Extensions.fs]: got you. in pr #384 fixed

**Comment by lust4life** [comment]
> It's just been a lot of xmas here, away from the laptop. Let me read through it.

sorry forget that, and 🎄  merry xmas & happy new year 🎊

**Pull request merged**